meilisearch-release-changelog/v1.9.0.md

```diff
@@ -0,0 +1,123 @@
+# v1.9.0 release changelogs
+
+<!-- The following line should ONLY be put PRE-release changelogs -->
+⚠️ Since this is a release candidate (RC), we do NOT recommend using it in a production environment. Is something not working as expected? We welcome [bug reports](https://github.com/meilisearch/meilisearch/issues/new/choose) and [feedback about new features](https://github.com/meilisearch/product/discussions).
+
+Meilisearch v1.9 includes performance improvements for hybrid search and the addition/updating of settings. This version benefits from multiple requested features, such as the new `frequency` matching strategy and the ability to retrieve similar documents.
+
+<!-- The following lines should NOT be put in the PRE-release changelogs -->
+🧰 All official Meilisearch integrations (including SDKs, clients, and other tools) are compatible with this Meilisearch release. Integration deployment happens between 4 to 48 hours after a new version becomes available.
+
+<!-- The following lines should NOT be put in the PRE-release changelogs -->
+Some SDKs might not include all new features. Consult the project repository for detailed information. Is a feature you need missing from your chosen SDK? Create an issue letting us know you need it, or, for open-source karma points, open a PR implementing it (we'll love you for that ❤️).
+
+# New features and updates 🔥
+
+## Hybrid search improvements
+
+Since we're focusing on AI innovation, this version introduces multiple improvements and changes related to hybrid search.
+More detailed changelog [here](https://meilisearch.notion.site/v1-9-AI-search-changes-e90d6803eca8417aa70a1ac5d0225697#38e6d3adf40e4ef1be14a3c4be39df94).
+
+Done by @dureuill and @irevoire in #4633 and #4649
+
+### ⚠️ Breaking changes of hybrid search usage
+
+- Before v1.9, an empty array in `_vectors.embedder` used to be interpreted as a single embedding of dimension 0 when specifying embeddings in documents. In v1.9 it is now interpreted as 0 embedding. The previous behavior was surprising and not useful.
+- From v1.9, by default, `_vectors` will not be returned anymore during a search. This will considerably improve performance. To display it, a new boolean search parameter has been introduced: `retrieveVectors` (default: `false`).
+
+### Improvements
+
+Meilisearch v1.9.0 improves performance when indexing and using hybrid search, avoiding useless operations and optimizing the important ones.
+
+## Filter by score
+
+TBD
```

Comment:
````suggestion
To filter returned documents by their ranking score, a new `rankingScoreThreshold` parameter has been added to the search and similar routes.

When a `rankingScoreThreshold` is provided, the results of the search/similar request are modified in the following way:

1. No document whose `_rankingScore` is under the `rankingScoreThreshold` is returned
2. Any document encountered during the search that is under the threshold is removed from the set of candidates and won’t count towards the `estimatedTotalHits`, `totalHits` and the facet distribution.

### Examples

request with**out** score threshold:

```json
POST /indexes/movies/search
{
  "q": "Badman dark returns 1",
  "showRankingScore": true,
  "limit": 5
}
```

results:

```json
{
	"hits": [
	    {
	      "title": "Batman the dark knight returns: Part 1",
	      "id": "A",
	      "_rankingScore": 0.93430081300813
	    },
	    {
	      "title": "Batman the dark knight returns: Part 2",
	      "id": "B",
	      "_rankingScore": 0.6685627880184332
	    },
	    {
	      "title": "Badman",
	      "id": "E",
	      "_rankingScore": 0.25
	    },
	    {
	      "title": "Batman Returns",
	      "id": "C",
	      "_rankingScore": 0.11553030303030302
	    },
	    {
	      "title": "Batman",
	      "id": "D",
	      "_rankingScore": 0.11553030303030302
	    }
	],
	"query": "Badman dark returns 1",
	"processingTimeMs": 11,
	"limit": 5,
	"offset": 0,
	"estimatedTotalHits": 62
}
```

request **with** score threshold:

```json
POST /indexes/movies/search
{
  "q": "Badman dark returns 1",
  "showRankingScore": true,
  "limit": 5
  "rankingScoreThreshold": 0.2
}
```

results:

```json
{
	"hits": [
	    {
	      "title": "Batman the dark knight returns: Part 1",
	      "id": "A",
	      "_rankingScore": 0.93430081300813
	    },
	    {
	      "title": "Batman the dark knight returns: Part 2",
	      "id": "B",
	      "_rankingScore": 0.6685627880184332
	    },
	    {
	      "title": "Badman",
	      "id": "E",
	      "_rankingScore": 0.25
	    }
	],
	"query": "Badman dark returns 1",
	"processingTimeMs": 11,
	"limit": 5,
	"offset": 0,
	"estimatedTotalHits": 3
}
```

### Known limitations

⚠️ For performance reasons, if Meilisearch finds `limit` hits above the `rankingScoreThreshold`, then the ranking score of the remaining documents is not evaluated, and so they are not removed from the set of candidates, even if their ranking score would be below the threshold.

As a result, in this configuration the `estimatedTotalHits`, `totalHits` and the facet distribution may be overapproximation of their values.
````